File: Web Sol/numbers.js
let arrayOfNumbers = document.getElementById('array');
let target = document.getElementById('target');

let positive = document.getElementById('positive');
let negative = document.getElementById('negative');

let solveBtn = document.getElementById('solve');
let resetBtn = document.getElementById('reset');

function solveChallenge(){

    arrayOfNumbers = arrayOfNumbers.value.split(' ').map(Number);
    target = target.value;
    
    let found = false;

    for(let i=0; i<arrayOfNumbers.length;i++){
        if(!found){
            let tryNumber = arrayOfNumbers[i];
            let searched = target - tryNumber;
            for(let j=0; j<arrayOfNumbers.length; j++){
                if(arrayOfNumbers[j] === searched){
                    if(tryNumber === searched) continue;
                    positive.style.display = "block";
                    positive.innerText = 'First number is '+ tryNumber + ' found at position ' + i + ' and second number is ' + arrayOfNumbers[j] + ' found at position ' + j + '.';
                    found = true;
                    resetBtn.style.display = "block";
                    solveBtn.style.display = "none";
                    break;
                }            
            }
        }
    }

    if(!found){
        negative.style.display = "block";
        negative.innerText = 'Combination not found!';
        resetBtn.style.display = "block";
        solveBtn.style.display = "none";
    }

}

function resetChallenge(){
    location.reload();
}

File: numbers.cpp
#include <iostream>

using namespace std;

int main()
{
    int arraySize;
    cout<<"Set array size: ";
    cin>>arraySize;
    
    int arrayOfNumbers[arraySize],target;
    
    cout<<"Insert array:";
    for(int i=0; i<arraySize; i++)
        cin>>arrayOfNumbers[i];
    
    cout<<"Insert target: ";
    cin>>target;
    
    bool found = false;
    
    for(int i=0; i<arraySize;i++){
        if(!found){
            int tryNumber = arrayOfNumbers[i];
            int searched = target - tryNumber;
            for(int j=0; j<arraySize; j++){
                if(arrayOfNumbers[j] == searched){
                if(tryNumber == searched) 
                    continue;
                cout<<endl<<endl;
                cout<<"First number is "<<tryNumber<<" and it's position is "<< i<<"."<<endl;
                cout<<"Second number is "<<arrayOfNumbers[j]<<" and it's position is "<< j<<"."<<endl;
                found = true;
                break;
                }
            }
        }
    }
    
    if(!found)
        cout<<"Combination not found!"<<endl;

    return 0;
}
